xc/xenops_server_xen.ml

```diff
@@ -2160,6 +2160,35 @@ module VIF = struct
 			_locking_mode, "disabled";
 		]
 
+	let xenstore_of_static_ip_setting vif =
+		let constant_setting = [ "static-ip-setting/mac", vif.mac;
+					 "static-ip-setting/error-code", "0";
+					 "static-ip-setting/error-msg", "" ]
+		in
+		let ipv4_setting = match vif.ipv4_configuration with
+		| Vif.Unspecified4 -> [ "static-ip-setting/enabled", "0" ]
+		| Vif.Static4 (address, gateway) ->
+			let enabled = [ "static-ip-setting/enabled" , "1" ] in
+			let setting = match gateway with
+			| None -> [ "static-ip-setting/address", (List.hd address) ]
```

Comment:
Instead of doing `List.hd`, it is safer to use the pattern `address :: _` in the match, and separately handle the case where the list is empty (`Static4 ([], _)`).